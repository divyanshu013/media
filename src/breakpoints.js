const breakpoints = {
  xsmall: 420,
  small: 576,
  medium: 768,
  large: 992,
  xlarge: 1200,
};

export default breakpoints;
