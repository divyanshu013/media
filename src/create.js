const create = (breakpoints, type) =>
  Object.entries(breakpoints).reduce(
    (mediaObject, [breakpoint, value]) => ({
      ...mediaObject,
      [breakpoint]: `@media (${type}-width: ${value}px)`,
    }),
    {},
  );

export default create;
