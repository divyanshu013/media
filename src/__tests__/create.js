import create from '../create';
import breakpoints from '../breakpoints';

test('snapshot create min query', () => {
  const query = create(breakpoints, 'min');
  expect(query).toMatchSnapshot();
});

test('snapshot create max query', () => {
  const query = create(breakpoints, 'max');
  expect(query).toMatchSnapshot();
});
