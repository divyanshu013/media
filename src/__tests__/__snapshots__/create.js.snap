// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot create max query 1`] = `
Object {
  "large": "@media (max-width: 992px)",
  "medium": "@media (max-width: 768px)",
  "small": "@media (max-width: 576px)",
  "xlarge": "@media (max-width: 1200px)",
  "xsmall": "@media (max-width: 420px)",
}
`;

exports[`snapshot create min query 1`] = `
Object {
  "large": "@media (min-width: 992px)",
  "medium": "@media (min-width: 768px)",
  "small": "@media (min-width: 576px)",
  "xlarge": "@media (min-width: 1200px)",
  "xsmall": "@media (min-width: 420px)",
}
`;
